c7n/resources/elasticsearch.py

```diff
@@ -116,6 +116,105 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'EncryptionAtRestOptions.KmsKeyId'
 
 
+@ElasticSearchDomain.filter_registry.register('cross-cluster')
+class ElasticSearchSearchConnections(Filter):
+    """Check for Cloudfront distribution config values
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: elasticsearch-cross-cluster
+            resource: aws.elasticsearch
+            filters:
+              - type: cross-cluster
+                inbound:
+                    key: SourceDomainInfo.OwnerId
+                    value: 0123456789012
+                    op: contains
+                outbound:
+                    key: SourceDomainInfo.OwnerId
+                    value: 0123456789012
+                    op: contains
+   """
+
+    schema = type_schema(
+        'cross-cluster',
+        inbound={
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'key': {'type': 'string'},
+                'value_type': {'$ref': '#/definitions/filters_common/value_types'},
+                'default': {'type': 'object'},
+                'value_regex': {'type': 'string'},
+                'value_from': {'$ref': '#/definitions/filters_common/value_from'},
+                'value': {'$ref': '#/definitions/filters_common/value'},
+                'op': {'$ref': '#/definitions/filters_common/comparison_operators'}
+            }},
+        outbound={
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'key': {'type': 'string'},
+                'value_type': {'$ref': '#/definitions/filters_common/value_types'},
+                'default': {'type': 'object'},
+                'value_regex': {'type': 'string'},
+                'value_from': {'$ref': '#/definitions/filters_common/value_from'},
+                'value': {'$ref': '#/definitions/filters_common/value'},
+                'op': {'$ref': '#/definitions/filters_common/comparison_operators'}
+            }})
+    schema_alias = False
+    annotation_key = "c7n:SearchConnections"
+    annotate = False
+    permissions = ('es:ESCrossClusterGet',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('es')
+        results = []
+
+        for r in resources:
+            if self.annotation_key in r:
+                continue
```

Comment:
the annotation key here is for the matches not for the pulled in data, ie. which implies multiple uses of this filter in a policy end up being a no-op. the intent for the annotation key is to cache the data from the api call, not to stop matches. its common to have a separate annotation key for the actual matches.